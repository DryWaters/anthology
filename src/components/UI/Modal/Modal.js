import React, { Fragment } from "react";

import Backdrop from "../Backdrop/Backdrop";
import classes from "./Modal.module.scss";

const Modal = props => {
  const modalStyles = props.show ? [classes.show] : [classes.hide];
  modalStyles.push(classes.Modal);

  return (
    <Fragment>
      <Backdrop show={props.show} clicked={props.modalClose} />
      <div className={modalStyles.join(" ")}>{props.children}</div>
    </Fragment>
  );
};

export default Modal;
