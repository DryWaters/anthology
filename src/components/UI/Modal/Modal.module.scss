@import "../../../variables.scss";

.Modal {
  position: fixed;
  z-index: 500;
  background-color: $color-secondary;
  min-width: 30rem;
  width: 50%;
  box-shadow: 1px 1px 1px $color-dark;
  padding: 0.5rem;
  left: 50%;
  top: 50%;
  transition: all 0.3s ease-out;
}

.show {
  transform: translate(-50%, -50%);
  opacity: 1;
}

.hide {
  transform: translate(-50%, -100vh);
  opacity: 0;
}

@media (min-width: 800px) {
  .Modal {
    padding: 1.6rem;
  }
}
