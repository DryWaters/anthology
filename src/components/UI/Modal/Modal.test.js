import React from "react";
import { shallow } from "enzyme";

import Modal from "./Modal";

describe("<Modal />", () => {
  it("should be visible if props.show is passed in", () => {
    const wrapper = shallow(<Modal show modalClose={() => {}} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(".show")).toHaveLength(1);
  });

  it("should not be visible if props.show is false", () => {
    const wrapper = shallow(<Modal show={false} modalClose={() => {}} />);
    expect(wrapper.find(".hide")).toHaveLength(1);
  });

  it("should render children if passed in", () => {
    const child = <div className="someText">Some Text</div>;
    const wrapper = shallow(
      <Modal show modalClose={() => {}}>
        {child}
      </Modal>
    );
    expect(wrapper.contains(child)).toEqual(true);
  });
});
