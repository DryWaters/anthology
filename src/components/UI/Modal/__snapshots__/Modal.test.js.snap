// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> should be visible if props.show is passed in 1`] = `ShallowWrapper {}`;
